### ISSUE TITLE
`nextcloud.existingSecret` doesn't cause `NEXTCLOUD_AUTH_TOKEN` to be set

### ISSUE THREAD

**Issue by felixscheinost** [nextcloud/helm#508]
## Describe your Issue

There is a `or .Values.nextcloud.existingSecret.enabled` missing here: https://github.com/nextcloud/helm/blob/3b7b2f24ca5c8da56b4ad29b80e66ded4a6da69a/charts/nextcloud/templates/metrics-deployment.yaml#L41

**Comment by jessebot** [comment]
That makes sense! You can submit a PR for this, just make sure the bump the `version` in Chart.yaml and this seems like an easy approval and merge :) If you, or others, submit a PR for this, please link back to this issue by including `closes #508` in the PR description. Further details about contributing can be found [here](https://github.com/nextcloud/helm/blob/main/CONTRIBUTING.md#pull-requests).

**Comment by jessebot** [comment]
It might actually make sense to introduce a values.yaml change here as well 🤔 Here's what I'm thinking:

We change this line in the `values.yaml`:
https://github.com/nextcloud/helm/blob/bf6cc4a9df0b3bffd3915dc940ddbec71976429e/charts/nextcloud/values.yaml#L86

To this:

```yaml
 tokenKey: ""
```

We then change these lines in `charts/nextcloud/templates/metrics/deployment.yaml`:
https://github.com/nextcloud/helm/blob/bf6cc4a9df0b3bffd3915dc940ddbec71976429e/charts/nextcloud/templates/metrics/deployment.yaml#L43-L48

To:
```go
  {{- if or .Values.metrics.token .Values.nextcloud.existingSecret.tokenKey }} 
  - name: NEXTCLOUD_AUTH_TOKEN 
   valueFrom: 
     secretKeyRef: 
       name: {{ .Values.nextcloud.existingSecret.secretName | default (include "nextcloud.fullname" .) }} 
       key: {{ .Values.nextcloud.existingSecret.tokenKey | default "tokenKey" }} 
```

This way, we only use `.Values.nextcloud.existingSecret.tokenKey` if it exists, but if we create the secret ourselves, we still default back to `tokenKey` for our own secret. Otherwise, we can still use the plain text `.Values.metrics.token`.

**Comment by jessebot** [comment]
After #508 was merged, you now need to specify a `tokenKey`, but this issue should be fixed now. If it's still broken for whatever reason, please feel free to open a new issue with the latest errors or issues you're noticing, or comment on this one, and we can reopen it. Thank you for reporting the issue!
